compiler/rustc_typeck/src/check/fn_ctxt/checks.rs

```diff
@@ -799,20 +799,25 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 Error::Invalid(provided_idx, expected_idx, compatibility) => {
                     let (formal_ty, expected_ty) = formal_and_expected_inputs[expected_idx];
                     let (provided_ty, provided_span) = provided_arg_tys[provided_idx];
-                    if let Compatibility::Incompatible(error) = &compatibility {
+                    if let Compatibility::Incompatible(Some(_)) = &compatibility {
                         let cause = &self.misc(provided_span);
                         let trace = TypeTrace::types(cause, true, expected_ty, provided_ty);
-                        if let Some(e) = error {
-                            self.note_type_err(
-                                &mut err,
-                                &trace.cause,
-                                None,
-                                Some(trace.values),
-                                e,
-                                false,
+                        self.note_type_err(
+                            &mut err,
+                            &trace.cause,
+                            None,
+                            Some(trace.values),
+                            // We actually forgo the TypeError we collect during `demand_compatible`,
+                            // since we're always going to note the type error using the argument span
+                            // and reporting a nested type's mismatch is not really helpful.
+                            &ty::error::TypeError::Sorts(ty::error::ExpectedFound::new(
```

Comment:
can you import both `TypeError` and `ExpectedFound` instead?

another nit: please move the construction of the new `TypeError` and the comment out of the argument list, that's easier to read imo